src/devices/cellular.c

```diff
@@ -0,0 +1,534 @@
+/*
+ * Race Capture Firmware
+ *
+ * Copyright (C) 2016 Autosport Labs
+ *
+ * This file is part of the Race Capture firmware suite
+ *
+ * This is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details. You should
+ * have received a copy of the GNU General Public License along with
+ * this code. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "LED.h"
+#include "api.h"
+#include "array_utils.h"
+#include "capabilities.h"
+#include "cellular.h"
+#include "cell_pwr_btn.h"
+#include "constants.h"
+#include "cpu.h"
+#include "dateTime.h"
+#include "gsm.h"
+#include "loggerConfig.h"
+#include "mod_string.h"
+#include "printk.h"
+#include "serial_buffer.h"
+#include "serial.h"
+#include "sara_u280.h"
+#include "sim900.h"
+#include "taskUtil.h"
+
+#include <stdbool.h>
+
+#define AUTH_MAX_JSMN_TOKENS	5
+
+static const struct cell_modem_methods* methods;
+static const struct at_config *at_cfg;
+
+static struct cellular_info cell_info;
+static struct telemetry_info telemetry_info;
+static const enum log_level serial_dbg_lvl = TRACE;
+
+static const struct at_config* get_safe_at_config()
+{
+        /* Safe but un-optimal values for all modems */
+        static const struct at_config cfg = {
+                .urc_delay_ms = 250,
+        };
+
+        return &cfg;
+}
+
+static void cell_serial_tx_cb(const char *data)
+{
+        static bool pr_tx_pfx = true;
+
+        for(; *data; ++data) {
+                if (pr_tx_pfx) {
+                        printk(serial_dbg_lvl, "[cell] tx: ");
+                        LED_toggle(0);
```

Comment:
No.  Right now this is a callback method called by the serial library so I can see data on a singular serial bus.